pkg/agent/flowexporter/connections/connections.go

```diff
@@ -122,18 +122,25 @@ func (cs *connectionStore) addOrUpdateConn(conn *flowexporter.Connection) {
 	cs.mutex.Lock()
 	defer cs.mutex.Unlock()
 	if exists {
+		// no need to update the connection when tcp state is TIME_WAIT
+		if existingConn.TCPState == "TIME_WAIT" {
+			return
+		}
 		// Update the necessary fields that are used in generating flow records.
 		// Can same 5-tuple flow get deleted and added to conntrack table? If so use ID.
 		existingConn.StopTime = conn.StopTime
 		existingConn.OriginalBytes = conn.OriginalBytes
 		existingConn.OriginalPackets = conn.OriginalPackets
 		existingConn.ReverseBytes = conn.ReverseBytes
 		existingConn.ReversePackets = conn.ReversePackets
-		existingConn.IsPresent = true
 		// Reassign the flow to update the map
 		cs.connections[connKey] = *existingConn
 		klog.V(4).Infof("Antrea flow updated: %v", existingConn)
 	} else {
+		// avoid adding the connection that is about to close and has been removed before
+		if conn.IsRemoved {
```

Comment:
can't we look at `TIME_WAIT`, `CLOSE` and dying status flags here as well to identify those connections?

Reply:
Here is a little bit different, I need to make sure the flow record has been sent. Because if the connection comes with `TIME_WAIT` state, we will not be able to add and send the record. Please refer to the latest commit. Thanks!